Gems/Multiplayer/Code/Source/Pipeline/NetworkSpawnableHolderComponent.cpp

```diff
@@ -28,18 +30,40 @@ namespace Multiplayer
 
     void NetworkSpawnableHolderComponent::Activate()
     {
+        const auto agentType = GetMultiplayer()->GetAgentType();
+        const bool shouldSpawnNetEntities =
+            (agentType == MultiplayerAgentType::ClientServer || agentType == MultiplayerAgentType::DedicatedServer);
+
+        if(shouldSpawnNetEntities)
+        {
+            AZ::Transform rootEntityTransform = AZ::Transform::CreateIdentity();
+
+            AzFramework::TransformComponent* rootEntityTransformComponent =
```

Comment:
Transform is the only component you _don't_ need to find at the moment, since it's exposed through the TransformInterface.

Reply:
Good point, changed to `AZ::TransformBus::EventResult(rootEntityTransform, GetEntityId(), &AZ::TransformInterface::GetWorldTM);` since `Entity::GetTransform()` is deprecated and actually not set here yet.

It is set after the entire entity finishes activating, so still nullptr here..

Interestingly FindComponent isn't too bad here because TransformComponent is _almost_ always the first one in the list!
This is because Editor inserts it automatically as the first one when a new entity is created and many components depend on it and the sorting routine places it first.
So although FindComponent is O(N), in practice for TransformComponent it is like finding the smallest element in a sorted array - O(1). 

But obviously it's not guaranteed in general, so best to use TransformBus here.